### PR TITLE
Minim

### PR DESCRIPTION
**Pull request by munetoshi**
Could you confirm you can compile double brace for std::array definition with mayah's environment?
(cpu/munetoshi/ai.cc ll.23-30)

### CODE REVIEW

File: src/cpu/munetoshi/ai.cc [hunk 1]
```diff
@@ -14,19 +14,21 @@
 
 #include "evaluator.h"
 
+namespace {
+template<int N>
 munetoshi::grade inner_product(
-        const munetoshi::grade* vect1,
-        const munetoshi::grade* vect2,
-        int n);
+        const std::array<munetoshi::grade, N> vect1,
```
Comment (mayah) [inline src/cpu/munetoshi/ai.cc]: Why not const-ref? Unnecessary copy will occur.

File: src/cpu/munetoshi/ai.cc [hunk 2]
```diff
@@ -14,19 +14,21 @@
 
 #include "evaluator.h"
```
Comment (mayah) [inline src/cpu/munetoshi/ai.cc]: #include <algorithm> needed.

Comment (mayah) [inline src/cpu/munetoshi/ai.cc]: #include &lt;algorithm&gt; needed.

**Comment by mayah** [comment]
This patch itself looks having a few problems, but I merge it, and merge a follow-up patch.

File: src/cpu/munetoshi/ai.cc (hunk 1 above)
Comment (munetoshi) [inline src/cpu/munetoshi/ai.cc]: Oops, just a mistake. Will fix. Thx.

File: src/cpu/munetoshi/ai.cc (hunk 2 above)
Comment (munetoshi) [inline src/cpu/munetoshi/ai.cc]: Will do. thx.

**Pull request merged**